src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,32 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 public class FileWork {
+    private static final char SPECIFIED_CHARACTER = 'w';
+
     public String[] readFromFile(String fileName) {
-        return null;
+        List<String> listOfWords = new ArrayList<>();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fileName));
+            String line = reader.readLine();
+            while (line != null) {
+                for (String word : line.split(" ")) {
```

Comment:
```suggestion
                for (String word : line.toLowerCase().split("\\W+")) {
```
this way you can avoid calling toLowercase twice and  replaceAll()

Reply:
Roman, Do I need to put finally and close() even if try with resources is autocloseable?

Reply:
Sorry, it wasn't "try with recourses"... Now it is.

Reply:
if you use try with resourcec you don't need to close resources in finally, that's the beauty of it 🙂